### PR TITLE
[feature] Add W&B Eval Callback and W&B GradCAM Callback

### PR DESCRIPTION
**Pull request by ayulockin**
I have added two new opinionated Keras callbacks. The dir structure and API design is something we can discuss further but wanted to get it out crudely for the event in context.

I have loosely type hinted the classes. Do let me know if any quick fix is required. 

The callbacks were manually tested.

PS: The grad cam callback can be built by subclassing the eval callback. One can also argue to have the eval callback have one argument added to it for explainability bit. But thought of having two separate class (with not much code diff) for now. We can discuss the design more.

cc: @soumik12345

### CODE REVIEW

File: wandb_addons/callbacks/__init__.py
Comment (soumik12345) [inline wandb_addons/callbacks/__init__.py:1]: This module is not necessary.

File: wandb_addons/callbacks/keras/clf_eval_callback.py
Comment (soumik12345) [inline wandb_addons/callbacks/keras/clf_eval_callback.py:1]: This callback should accept a list of `tf.keras.metrics.Metric`, all of which should be evaluated and logged on the `wandb.Table`.

File: wandb_addons/callbacks/keras/gradcam.py
Comment (soumik12345) [inline wandb_addons/callbacks/keras/gradcam.py:1]: This is sick!!! 🔥

**Review by soumik12345 — CHANGES_REQUESTED** [review]
Hi @ayulockin 
The`WandbGradCAMCallback` looks great!!!
I have requested some changes with respect to `WandbGradCAMCallback`.

File: wandb_addons/callbacks/keras/clf_eval_callback.py
Comment (ayulockin) [reply wandb_addons/callbacks/keras/clf_eval_callback.py:1]: as per the discussion over WhatsApp, resolving this.

File: wandb_addons/callbacks/__init__.py
Comment (ayulockin) [reply wandb_addons/callbacks/__init__.py:1]: removed the __init__ file if that was what your meant

**Review by soumik12345 — APPROVED** [review]
